File: Organization-service/src/main/java/cl/privdata/organizationService/model/Person.java
package cl.privdata.organizationService.model;

import cl.privdata.organizationService.enums.DataStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "persons")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", updatable = false, nullable = false, columnDefinition = "UUID")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id")
    private Department department;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(name = "full_name", nullable = false, length = 200)
    private String fullName;

    @Column(name = "rut", length = 20)
    private String rut;

    @Column(name = "email", length = 150)
    private String email;

    @Column(name = "phone", length = 50)
    private String phone;

    @Column(name = "position", length = 120)
    private String position;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    //atributos para derecho cancelacion
    @Column(name = "blocked", nullable = false)
    private Boolean blocked;
    @Column(name = "anonymized", nullable = false)
    private Boolean anonymized;
    @Column(name = "deletion_request", nullable = false)
    private Boolean deletionRequest;
    @Enumerated(EnumType.STRING)
    @Column(name = "data_status", nullable = false)
    private DataStatus dataStatus;


}

File: Frontend-Privdata/src/types/person.ts
export type DataStatus = "ACTIVE" | "BLOCKED" | "DELETION_REQUESTED" | "PROCESSING_RESTRICTED" | "ANONYMIZED"

export interface Person {
  id: string
  organizationId: string
  departmentId: string | null
  departmentName: string | null
  firstName: string
  lastName: string
  fullName: string
  rut: string | null
  email: string | null
  phone: string | null
  position: string | null
  isActive: boolean
  createdAt: string
  updatedAt: string
  blocked: boolean
  anonymized: boolean
  deletionRequest: boolean
  dataStatus: DataStatus
}

export interface InvitePersonRequest {
  firstName: string
  lastName: string
  email: string
  rut: string
  position?: string
  departmentId?: string
  roleName: string
}

export interface UpdatePersonRequest {
  firstName: string
  lastName: string
  email?: string
  rut?: string
  phone?: string
  position?: string
  departmentId?: string
}

export interface InvitePersonResponse {
  person: Person
  user: {
    data: {
      userId: string
      email: string
      temporaryPassword: string
    }
  }
}
